tools/ci/flink-ci-tools/src/main/java/org/apache/flink/tools/ci/licensecheck/NoticeFileChecker.java

```diff
@@ -175,44 +177,32 @@ private static int checkNoticeFile(
         String moduleName = getModuleFromNoticeFile(noticeFile);
 
         // 1st line contains module name
-        List<String> noticeContents = Files.readAllLines(noticeFile);
+        NoticeContents noticeContents = NoticeParser.parseNoticeFile(noticeFile).orElse(null);
 
         final Map<Severity, List<String>> problemsBySeverity = new HashMap<>();
 
-        if (noticeContents.isEmpty()) {
+        if (noticeContents == null) {
             addProblem(problemsBySeverity, Severity.CRITICAL, "The NOTICE file was empty.");
         }
 
         // first line must be the module name.
-        if (!noticeContents.get(0).equals(moduleName)) {
+        if (!noticeContents.getNoticeModuleName().equals(moduleName)) {
```

Comment:
This is just as safe as the previous version: not at all.

The next commit refactors this bit.